### ISSUE TITLE
Dolt panics querying `dolt_history_$table`

### ISSUE THREAD

**Issue by ricardoreiter** [dolthub/dolt#6891]
Running this query in our DB crashes the sql-server:
```select * from dolt_history_sup_optionCondition where rowId = UUID_TO_BIN('0000bd29-d1ee-71c0-0430-ce35268a9ed3');```

```
2023-10-26 16:53:43 panic: runtime error: invalid memory address or nil pointer dereference
2023-10-26 16:53:43 [signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x1c1c254]
2023-10-26 16:53:43 
2023-10-26 16:53:43 goroutine 356 [running]:
2023-10-26 16:53:43 github.com/dolthub/dolt/go/libraries/doltcore/sqle/index.ordinalMappingFromIndex({0x30a3b20, 0xc00cbbe160})
2023-10-26 16:53:43     github.com/dolthub/dolt/go/libraries/doltcore/sqle/index/prolly_index_iter.go:139 +0x74
2023-10-26 16:53:43 github.com/dolthub/dolt/go/libraries/doltcore/sqle/index.newNonCoveringLookupBuilder(0xc00cbbe160?, 0xc003332480)
2023-10-26 16:53:43     github.com/dolthub/dolt/go/libraries/doltcore/sqle/index/index_lookup.go:308 +0x32c
2023-10-26 16:53:43 github.com/dolthub/dolt/go/libraries/doltcore/sqle/index.NewLookupBuilder(0x1?, {0x2ff1b38?, 0xc000b2c640?}, {0x30a3b20, 0xc00cbbe160}, {{0x5a, 0x8d, 0xaa, 0xb0, 0xb1, ...}}, ...)
2023-10-26 16:53:43     github.com/dolthub/dolt/go/libraries/doltcore/sqle/index/index_lookup.go:285 +0x805
2023-10-26 16:53:43 github.com/dolthub/dolt/go/libraries/doltcore/sqle.(*IndexedDoltTable).PartitionRows(0xc0111d4180, 0xca809f6a1ce0f4ee?, {0x2fe90e0, 0xc011565a40})
2023-10-26 16:53:43     github.com/dolthub/dolt/go/libraries/doltcore/sqle/indexed_dolt_table.go:85 +0x22d
2023-10-26 16:53:43 github.com/dolthub/dolt/go/libraries/doltcore/sqle.(*historyIter).Next(0xc0111d4200, 0xc002410140?)
2023-10-26 16:53:43     github.com/dolthub/dolt/go/libraries/doltcore/sqle/history_table.go:551 +0x131
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:65 +0x18a
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 ...4236 frames elided...
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0xc00c9cf540, 0xc00cb40000)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/table_iter.go:73 +0x265
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql/plan.(*FilterIter).Next(0xc00cbdc400, 0x2?)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/plan/filter.go:114 +0x39
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql/rowexec.(*projectIter).Next(0xc00c8b0240, 0x1?)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/rowexec/rel_iters.go:512 +0x29
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql/rowexec.transactionCommittingIter.Next(...)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/rowexec/transaction_iters.go:78
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/sql/plan.(*trackedRowIter).Next(0xc00c9cf580, 0xc00c8a3eb8?)
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/sql/plan/process.go:347 +0x24
2023-10-26 16:53:43 github.com/dolthub/go-mysql-server/server.(*Handler).doQuery.func2()
2023-10-26 16:53:43     github.com/dolthub/go-mysql-server@v0.17.1-0.20231024163156-64b6f6b4e8a6/server/handler.go:292 +0x125
2023-10-26 16:53:43 golang.org/x/sync/errgroup.(*Group).Go.func1()
2023-10-26 16:53:43     golang.org/x/sync@v0.3.0/errgroup/errgroup.go:75 +0x56
2023-10-26 16:53:43 created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 335
2023-10-26 16:53:43     golang.org/x/sync@v0.3.0/errgroup/errgroup.go:72 +0x96
```

**Comment by fulghum** [comment]
Thanks for the bug report. I was able to repro this and track down what's happening in the debugger. I've got a simpler repro test case coded up now: 
```sql
CREATE TABLE test (pk int PRIMARY KEY, v1 int);
INSERT INTO test VALUES (1,1), (4,4), (5,5);
call dolt_commit('-Am', 'adding table test');
alter table test modify column v1 varchar(100);
call dolt_commit('-am', 'changing non-pk col type in order to change tag');
SELECT * from dolt_history_test where pk=42;
```

Those statements generate the same error because Dolt isn't able to find an index that it thinks covers all the columns – it gets confused because one of the columns has a different "tag" (our internal column identifier) due to being modified. 

I'm not sure of the cleanest fix yet, but I've got a couple of ideas and will get a PR open soon to get some feedback from the team in case they have better ideas.
